remoteconfig_test.go

```diff
@@ -569,42 +566,108 @@ func (s *RemoteConfigSuite) TestdownloadJSONValidate() {
 	defer ts.Close()
 
 	c := &SampleConfig{}
-	err := DownloadJSONValidate(ts.URL, c)
+	err := LoadConfigFromURL(ts.URL, c)
 	assert.Nil(s.T(), err)
 }
 
-func (s *RemoteConfigSuite) TestdownloadJSONValidateErrorDownloadFailed() {
+func (s *RemoteConfigSuite) TestLoadConfigFromURL_NotOK() {
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		http.Error(w, "JSON Download Failed", http.StatusNotFound)
+		w.WriteHeader(http.StatusNotFound)
+		fmt.Fprintln(w, "Not Found")
 	}))
 	defer ts.Close()
 
+	c := &SQSQueueConfig{}
+	err := LoadConfigFromURL(ts.URL, c)
+	assert.NotNil(s.T(), err)
+	assert.Regexp(s.T(), regexp.MustCompile("Request to 'http://\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}:\\d{1,5}' returned non-200 OK status '404: Not Found'"), err.Error())
+}
+
+func (s *RemoteConfigSuite) TestLoadConfigFromURLError() {
+	c := &SQSQueueConfig{}
+	err := LoadConfigFromURL("invalid", c)
+	assert.NotNil(s.T(), err)
```

Comment:
`s.Error(err)` -- though I think you can probably omit this assertion in lieu of the one below.